src/Dumplie/Customer/Application/Query/Result/Cart.php

```diff
@@ -60,6 +83,6 @@ public function totalPrice() : float
      */
     public function currency() : string
     {
-        return $this->isEmpty() ? null : $this->items[0]->currency();
+        return $this->isEmpty() ? '' : $this->items[0]->currency();
```

Comment:
Then maybe instead of misleading exception about returned type we should throw something more descriptive `new CartException::noCurrencySet()` ?